Automattic/wp-calypso
# Calypso UI Components: DateRange: add customTitle

## Proposed Changes

* This PR updates the Calypso DateRange component to introduce an optional custom title

## Why are these changes being made?


* To extend the Calypso DateRange picker component for use with the Stats Date Control update.

## Testing Instructions



* open Calypso live branch
* navigate to `/devdocs/design/date-range`
* click on the last demo `With Custom Title` to open the calendar popover
* check the that the custom title displays
* check some other demos without the custom title and ensure the regular title isn't affected
* also thoroughly check on mobile and small displays

![image](https://github.com/user-attachments/assets/eb2f8ecd-cbc7-46fb-ab9a-f8b56d9dca54)

## Pre-merge Checklist
